src/rom6.c

```diff
@@ -121,14 +121,14 @@ bool8 SetUpFieldMove_RockSmash(void)
     if(ShouldDoBrailleStrengthEffect())
     {
         gSpecialVar_Result = GetCursorSelectionMonId();
-        gUnknown_03005DB0 = FieldCallback_Teleport;
-        gUnknown_0203CEEC = sub_8179834;
+        gFieldCallback2 = FieldCallback_PrepareFadeInFromMenu;
+        gPostMenuFieldCallback = sub_8179834;
         return TRUE;
     }
-    else if (npc_before_player_of_type(0x56) == TRUE)
+    else if (SetLastTalkedObjectInFrontOfPlayer(0x56) == TRUE)
```

Comment:
Use define

Reply:
done

Reply:
Rename `SetLastTalkedObjectInFrontOfPlayer` to `CheckObjectGraphicsInFrontOfPlayer`

Reply:
Egg pls